src/Orchard/Localization/LocalizationModule.cs

```diff
@@ -20,18 +22,23 @@ public class LocalizationModule : Module {
         protected override void AttachToComponentRegistration(IComponentRegistry componentRegistry, IComponentRegistration registration) {
 
             var userProperty = FindUserProperty(registration.Activator.LimitType);
-
             if (userProperty != null) {
-                var scope = registration.Activator.LimitType.FullName;
-
-                registration.Activated += (sender, e) => {
-                    if (e.Instance.GetType().FullName != scope) {
-                        return;
-                    }
-
-                    var localizer = _localizerCache.GetOrAdd(scope, key => LocalizationUtilities.Resolve(e.Context, scope));
-                    userProperty.SetValue(e.Instance, localizer, null);
-                };
+                List<string> scopes = new List<string>();
+                var type = registration.Activator.LimitType;
+                while (!type.Namespace.Equals("System")) {
```

Comment:
Add a comment, I assume it's because we don't need this behavior on CLR types, so that's an optimization